File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.java
class Solution {
    public ListNode swapNodes(ListNode head, int k) {
        int n = 0;
        
        ListNode h1 = head;
        
        while(h1 != null){
            n++;
            h1 = h1.next;
        }
        
        h1 = head;
        int curr;
        int k1 = k;
        
        while(k1-->1)
            h1 = h1.next;
        
        curr = h1.val;
        
        h1 = head;
        
        k1 = k;
        
        int temp;
        
        k1 = n-k1+1;
        
        while(k1-->1){
            h1 = h1.next;
        }
        
        temp = h1.val;
        h1.val = curr;
        
        k1 = k;
        h1 = head;
        
        while(k-->1)
            h1 = h1.next;
        
        h1.val = temp;
        
        
        return head;
    }
}

File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    
    int getLen(ListNode* head)
    {
        int len=0;
        while(head)
        {
            head=head->next;
            len++;
        }
        return len;
    }
    ListNode* swapNodes(ListNode* head, int k) {
        
        int len=getLen(head);
        int ind=1;
        ListNode* firstNode , *secondNode , *temp=head;
        
        while(temp)
        {
            if(firstNode!=NULL && secondNode!=NULL)
                break;
            if(ind==k)
                firstNode=temp;
            if(ind==len-k+1)
                secondNode=temp;
            
            temp = temp->next;
            ind++;
        }
        
        swap(firstNode->val , secondNode->val);
        
        return head;
        
    }
};